src/vivarium_cluster_tools/psimulate/worker/vivarium_work_horse.py

```diff
@@ -203,3 +223,54 @@ def format_and_record_details(
                 df.insert(df.shape[1] - 1, col_name, val)
         finished_results_metadata[key] = val
     return finished_results_metadata
+
+
+def run(sim: SimulationContext, job_parameters: JobParameters):
+    backup_freq = job_parameters.backup_configuration["backup_freq"]
+    if backup_freq:
+        backup_dir = job_parameters.backup_configuration["backup_dir"]
+        backup_path = (backup_dir / str(get_current_job().id)).with_suffix(".pkl")
+        time_to_save = time() + backup_freq
+        while not sim.past_stop_time():
+            sim.step()
+            if time() >= time_to_save:
+                sim.write_backup(backup_path)
+                time_to_save = time() + backup_freq
+    else:
+        sim.run()
+
+
+def get_backup(job_parameters: JobParameters) -> Optional[SimulationContext]:
+    backup_dir = job_parameters.backup_configuration["backup_dir"]
+    if not backup_dir.exists() or not os.listdir(backup_dir):
```

Comment:
Why do you need to check both?

Reply:
The backup directory doesn't get created at all if we're not making backups, and `os.listdir()` does a FileNotFoundError if the dir does not exist. I guess you could incorporate that into the try except block used later, but it seemed a bit convoluted to do the AFNP style thing in this case

Reply:
I'm not totally following why not to put this all inside the try/except below. That seems simpler to me and I don't see the downside. 

Also, shouldn't the directory always exist if the backup metadata file exists?